hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/ErrorResponse.java

```diff
@@ -2,27 +2,34 @@
 
 import java.util.Optional;
 
+import org.apache.commons.lang3.StringUtils;
+
 import ca.bc.gov.hlth.hnsecure.parsing.Util;
 
 public class ErrorResponse extends ResponseSegment {
-	
+
 	private static final String segmentIdentifier = "MSA";
-		
+
 	private static final String ackKnowledgementCode = "AR";
 
 	@Override
 	public String constructResponse(HL7Message messageObj, ErrorMessage errorMessage) {
 		return constructMSH(messageObj) + constructMSA(messageObj.getMessageControlId(), errorMessage);
 	}
-	
+
 	public String constructMSA(String messageControlID, ErrorMessage errorMessage) {
 		StringBuilder sb = new StringBuilder(segmentIdentifier);
 		sb.append(Util.HL7_DELIMITER);
 		sb.append(ackKnowledgementCode);
 		sb.append(Util.HL7_DELIMITER);
 		sb.append(Optional.ofNullable(messageControlID).orElse(""));
 		sb.append(Util.HL7_DELIMITER);
-		sb.append(errorMessage.getErrorSequence() + "  " + errorMessage.getErrorMessage());
+		if (StringUtils.isNotBlank(errorMessage.getFieldValue())) {
+			sb.append(errorMessage.getErrorSequence() + "  " + errorMessage.getErrorMessage() + " "
+					+ errorMessage.getFieldValue());
+		} else {
+			sb.append(errorMessage.getErrorSequence() + "  " + errorMessage.getErrorMessage());
+		}
```

Comment:
This may not apply as we need to change our approach but in case like this you can leave the sb.append(errorMessage.getErrorSequence() + "  " + errorMessage.getErrorMessage()) outside of the block and just have an if block to add the additional text. In the current version that code gets repeated.